azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -49,32 +78,25 @@ def dataframe_from_result_table(table: "Union[KustoResultTable, KustoStreamingRe
     columns = [col.column_name for col in table.columns]
     frame = pd.DataFrame(table.raw_rows, columns=columns)
 
-    # fix types
+    # converters_by_type overrides the default
+    if converters_by_type and not nullable_bools:
```

Comment:
Ohad mentioned that creating a new dict is wasteful, and I have to agree.

So checking them one after the other is probably the correct way.
Sorry fot the back and forth.